src/webgl2_render_pass.rs

```diff
@@ -145,11 +147,8 @@ impl<'a> RenderPass for WebGL2RenderPass<'a> {
     }
 
     fn draw_indexed(&mut self, indices: Range<u32>, _base_vertex: i32, instances: Range<u32>) {
-        let (index_type, type_size) = match self.pipeline_descriptor.as_ref().unwrap().index_format
```

Comment:
let's remove WebGL2RenderPass.pipeline_descriptor, as it was used only for providing access to index_format.
We can store gl index_format / index_type_size in WebGL2Pipeline object (in set_index_buffer method) or directly in WebGL2RenderPass object

Reply:
I had attempted moving the `IndexFormat` to `WebGL2Pipeline` at some point, but the code ended up in a state where I had duplicated this chunk of code from `setup_vao`

```
        let resources = &self.render_context.render_resource_context.resources;
        let mut pipelines = resources.pipelines.write();
        let pipeline_handle = self.pipeline.as_ref().unwrap();
        let pipeline = pipelines.get_mut(&pipeline_handle).unwrap();
```

And that bothered me. And even then I never got it to a point where things even showed up on the screen again so I gave up.

I'll give it another go though.

Reply:
Ah yes, this is the deeply mysterious panic I was getting when I attempted this last time:

```
wasm.js:752 panicked at 'Parking not supported on this platform', /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/parking_lot_core-0.8.2/src/thread_parker/wasm.rs:26:9
```

Reply:
Okay, I think I worked around that. Commit is incoming. But maybe you can suggest a cleaner way to do it.

Reply:
I think it is fine now, it may be refactored later if we find a better way.